static/app/components/replays/useReplaysCount.tsx

```diff
@@ -46,78 +49,52 @@ function useReplaysCount({
         'Unable to query both groupId and transactionName simultaneously in useReplaysCount()'
       );
     }
+
     if (groupIds && groupIds.length) {
-      const groupsToFetch = toArray(groupIds).filter(gid => !(gid in replayCounts));
-      if (!groupsToFetch.length) {
-        return null;
+      const groupsToFetch = filterUnseen(groupIds);
+      if (groupsToFetch.length) {
+        return {
+          field: 'issue.id' as const,
+          conditions: `issue.id:[${groupsToFetch.join(',')}]`,
+        };
       }
-
-      return {
-        field: 'issue.id' as const,
-        conditions: `issue.id:[${groupsToFetch.join(',')}]`,
-      };
+      return null;
     }
+
     if (transactionNames && transactionNames.length) {
-      return {
-        field: 'transaction' as const,
-        conditions: `event.type:transaction transaction:[${toArray(transactionNames).join(
-          ','
-        )}]`,
-      };
+      const txnsToFetch = filterUnseen(transactionNames);
+      if (txnsToFetch.length) {
+        return {
+          field: 'transaction' as const,
+          conditions: `event.type:transaction transaction:[${txnsToFetch.join(',')}]`,
+        };
+      }
+      return null;
     }
     return null;
-  }, [replayCounts, groupIds, transactionNames]);
-
-  const eventView = useMemo(
-    () =>
-      EventView.fromSavedQuery({
-        id: '',
-        name: `Errors within replay`,
-        version: 2,
-        fields: ['count_unique(replayId)', String(query?.field)],
-        query: `!replayId:"" ${query?.conditions}`,
-        projects: projectIds,
-      }),
-    [projectIds, query]
-  );
+  }, [filterUnseen, groupIds, transactionNames]);
 
   const fetchReplayCount = useCallback(async () => {
     try {
       if (!query) {
         return;
       }
 
-      if (query.field === 'issue.id') {
-        const response = await api.requestPromise(
-          `/organizations/${organization.slug}/issue-replay-count/`,
-          {
-            query: {
-              query: query.conditions,
-              statsPeriod: '14d',
-              project: projectIds,
-            },
-          }
-        );
-        setReplayCounts({...zeroCounts, ...response});
-      } else {
-        const [data] = await doDiscoverQuery<TableData>(
-          api,
-          `/organizations/${organization.slug}/events/`,
-          eventView.getEventsAPIPayload({query: {}} as Location<any>)
-        );
-
-        const counts = data.data.reduce((obj, record) => {
-          const key = record[query.field] as string;
-          const val = record['count_unique(replayId)'] as number;
-          obj[key] = val;
-          return obj;
-        }, zeroCounts);
-        setReplayCounts(counts);
-      }
+      const response = await api.requestPromise(
+        `/organizations/${organization.slug}/replay-count/`,
+        {
+          query: {
+            query: query.conditions,
+            statsPeriod: '14d',
```

Comment:
Do we have a ticket to look into this? I think having this as a static 14 days will end up causing some confusion down the road.